crates/worktree/src/ignore.rs

```diff
@@ -84,6 +100,29 @@ impl IgnoreStack {
                     ignore::Match::Whitelist(_) => false,
                 }
             }
+            IgnoreStackEntry::RepoExclude { ignore, parent } => {
+                let combined_path;
+                let abs_path = if let Some(repo_root) = self.repo_root.as_ref() {
+                    combined_path = ignore.path().join(
+                        abs_path
+                            .strip_prefix(repo_root)
+                            .expect("repo root should be a parent of matched path"),
+                    );
+                    &combined_path
+                } else {
+                    abs_path
+                };
+
+                match ignore.matched(abs_path, is_dir) {
+                    ignore::Match::None => IgnoreStack {
+                        repo_root: self.repo_root.clone(),
+                        top: parent.clone(),
+                    }
+                    .is_abs_path_ignored(abs_path, is_dir),
+                    ignore::Match::Ignore(_) => true,
+                    ignore::Match::Whitelist(_) => false,
+                }
+            }
```

Comment:
I think this can be simplified a bit, and we can just do `ignore.matched(abs_path)`.

Reply:
For last two match arms? I'm not sure how `ignore.matched(abs_path)` would fit, do you mean something like:

```rs
match ignore.matched(abs_path, is_dir) {
    ignore::Match::None => IgnoreStack {
        repo_root: self.repo_root.clone(),
        top: parent.clone(),
    }
    .is_abs_path_ignored(abs_path, is_dir),
    matched => matched.is_ignore(),
}
```